### ISSUE TITLE
Plugins crash (3.2.3 / 3.2.5) with at failed assertion "slang_emit.c:570: emit_compare" (Ubuntu 18.04)

### ISSUE THREAD

**Issue by edwardchapin** [michaelwillis/dragonfly-reverb#99]
I've been experimenting with your beautiful sounding plugins using Audacity on my Ubuntu 18.04 laptop since ~January.

Recently I tried them out with Ardour 6 (ALSA), and it would immediately crash when I tried to load them:
```
ardour-6.6.0: shader/slang/slang_emit.c:570: emit_compare: Assertion `n->Children[0]->Store->Size == n->Children[1]->Store->Size' failed.
Aborted (core dumped)
```

I then tried running a Jack server (via `qjackctl`) and then starting the stand-alone effects (`DragonflyEarlyReflections` etc.) and they crashed in exactly the same way (all four of them).

Initially I was running 3.2.3 built from source, so I just tried building 3.2.5 from source (failed in the same way), and also trying out your pre-built 3.2.5 tarball (also failed in the same way):
```
$ ./DragonflyRoomReverb
DragonflyRoomReverb: shader/slang/slang_emit.c:570: emit_compare: Assertion `n->Children[0]->Store->Size == n->Children[1]->Store->Size' failed.
Aborted (core dumped)
```

They continue to work with Audacity.

**Comment by michaelwillis** [comment]
@edwardchapin What video hardware are you running? Some people have reported that Dragonfly Reverb crashes when they are using their video card, but if they switch to using the integrated graphics it doesn't crash. Admittedly that is just a workaround, and it would be better to fix the problem, but I still don't have enough insight into why it crashes to fix it.

**Comment by edwardchapin** [comment]
I'm already using the integrated graphics adapter:
```
$ lspci | grep VGA
00:02.0 VGA compatible controller: Intel Corporation Haswell-ULT Integrated Graphics Controller (rev 09)
```

I'm game to dig a little deeper and do a debugging session (e.g., compile with `-g` and run one of the plugins in vscode) if that's potentially of use.

**Comment by edwardchapin** [comment]
Ahh... it was worth running in the debugger. It was indeed an error buried way down in libGL. I was surprised to see my IDE had the source file (from mesa3d) when I looked through the call stack, and I realized it was picking up an ancient locally-built libGL.so from a project I worked on years ago! (still had dev environment in my .bashrc).

:man_facepalming:

I got rid of that and all is well... sorry for the false alarm.
